### DISCUSSION TITLE
exponential function solving problem

### DISCUSSION THREAD

**Discussion by dancimanc** [wxMaxima-developers/wxmaxima#1901] [Q&A]
Hello All!
I am trying to solve an equation for "Bg", but I can't get any actual results. Can you help me what do I do wrong?

![image](https://github.com/wxMaxima-developers/wxmaxima/assets/65621200/dac039d2-7f69-4349-a140-3eb7250b63a6)

It has this shape, idk what is the problem...
![image](https://github.com/wxMaxima-developers/wxmaxima/assets/65621200/34fb316c-5bd7-4752-baf4-395410839f3e)

Thank you for the help!

**Comment by gunterkoenigsmann** [comment]
I believe that if this has a symbolic solution that solution will be complicated and long. But find_root or mnewton should be able to find a numeric solution.

**Reply by dancimanc** [reply]
I will try, thanks. But if you check the 2nd picture you can see it has 2 real solutions (around 0.048 and -0.048).

**Comment by gunterkoenigsmann** [comment]
Do you have exact numbers for the floats? Maxima is much stronger with exact numbers than with floats and your float won't hit the exact points you want them to. And in such cases I often try to bring the equation into a form Maxima will be able to solve by doing something like %/denim(RHS(%)) and similar. But this time the equation will end up as something similar to exp(x)=k*x. That afaik is only symbolically solvable with a bit of luck.

**Comment by dancimanc** [comment]
Sorry, this is the actual function plotted
![image](https://github.com/wxMaxima-developers/wxmaxima/assets/65621200/bf9b8913-b193-4b54-b469-0c2603494100)
 
the previous version is a simplified approximate for the function

**Reply by dancimanc** [reply]
x would be Bg and you can see that it should have a solution for y = 1
I am not sure how to bring this to a form where maxima can solve it... I can't get x (Bg) out from argument of exp, neither log..

**Comment by daute** [comment]
Hi,
first I create a function, `f(Bg)`, where I want to find the roots:

```
(%i1) f(Bg):=1.10111380032118*%e^(-33*Bg^2)/(7.4*Bg^2+1)-1;
                                                       2
                                              (- 33) Bg
                           1.10111380032118 %e
(%o1)             f(Bg) := ----------------------------- - 1
                                          2
                                    7.4 Bg  + 1
```

(By subtracting 1 on both sides). Now I can solve that for `f(Bg)=0`. An exact solution is most likely not possible (I used (probably not exact) floats when defining my function), but one can get numeric approximations. From the plot you saw, that one solution is between -1 and 0 and one solution between 0 and 1.
The sign of `f(Bg)` is different at the end points of the interval, so one can use the interval method to solve it:

```
(%i2) find_root(f,Bg,-1,0);
(%o2)                        - 0.04886750570984203
(%i3) find_root(f,Bg,0,1);
(%o3)                         0.04886750570984203
```

One can also try the newton method, but there one must be careful when selecting the start value:
```
(%i4) load(newton1);
(%o4)        /usr/share/maxima/5.43.2/share/numeric/newton1.mac
(%i5) newton(f(Bg),Bg,0.02,0.0001);
(%o5)                        0.0488675116779828
```

Best regards, Wolfgang

(I used command line Maxima here, it is easier to cut&paste a formula as text, than only having a graphic where one must type the formula character by character.
And the question is not an issue with wxMaxima (the graphical frontend).
